pilot/cmd/istioctl/inject.go

```diff
@@ -18,32 +18,64 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"os"
+
 	// TODO(nmittler): Remove this
+	"github.com/ghodss/yaml"
 	_ "github.com/golang/glog"
 	"github.com/spf13/cobra"
-	"k8s.io/api/core/v1"
 
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/cmd"
+	"istio.io/istio/pilot/model"
 	"istio.io/istio/pilot/platform/kube"
 	"istio.io/istio/pilot/platform/kube/inject"
 	"istio.io/istio/pkg/log"
 	"istio.io/istio/pkg/version"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const configMapKey = "mesh"
+
+func getMeshConfigFromConfigMap(kubeconfig string) (*meshconfig.MeshConfig, error) {
```

Comment:
I'm a bit surprised we don't already have that code elsewhere in pilot pkgs?

Reply:
We did (see https://github.com/istio/istio/blob/master/pilot/platform/kube/inject/configmap.go) but I removed it. I think it's preferable to access the MeshConfig via file and rely on the deployment to mount the ConfigMap as a volume. This decouples MeshConfig ingestion from k8s configuration. 

I added a refactored version of GetMeshConfig() back as a private function here so that `kube-inject` could retain backwards compatibility with the previous version which required access to a running k8s cluster to inject resources. The new version does not have such a requirement and the MeshConfig can be loaded from a file (see `--meshConfigFile`). Other components similarly access the configuration by file, e.g. https://github.com/istio/istio/blob/master/pilot/cmd/pilot-discovery/main.go#L82